src/core/qgsmaplayerregistry.cpp

```diff
@@ -203,14 +220,17 @@ void QgsMapLayerRegistry::onMapLayerDeleted( QObject* obj )
 
 QMap<QString, QgsMapLayer*> QgsMapLayerRegistry::mapLayers()
 {
-  return mMapLayers;
+  QReadLocker lock( &mRegistryLock );
+
+  QMap<QString, QgsMapLayer*> mapLayers = mMapLayers;
+  mapLayers.detach();
+  return mapLayers;
```

Comment:
Would this be needed? Isn't the first thing the container does before a modification is detach anyway?